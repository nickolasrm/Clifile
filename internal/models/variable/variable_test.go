package variable_test

import (
	"testing"

	"github.com/nickolasrm/clifile/internal/models/variable"
	. "github.com/nickolasrm/clifile/test/util"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

func TestVariable(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Variable Suite")
}

var _ = Describe("Variable", func() {
	When("a new variable is created", func() {
		var v *variable.Variable

		BeforeEach(func() {
			v = variable.NewVariable("name", "value")
		})

		It("should return a variable type", func() {
			Expect(v).ToNot(BeNil())
		})
		When("Name() is called", func() {
			It("should return the variable name", func() {
				MatchSnapshot(v.Name())
			})
		})
		When("Value() is called", func() {
			It("should return its value", func() {
				MatchSnapshot(v.Value())
			})
		})
		When("SetValue() is called", func() {
			It("should update the variable value", func() {
				v.SetValue("custom")
				MatchSnapshot(v.Value())
			})
		})
	})
})
