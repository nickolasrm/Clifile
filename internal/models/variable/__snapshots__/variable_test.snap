
[Variable when a new variable is created when Name() is called should return the call name - 1]
[]interface {}{
    "name",
}
---

[Variable when a new variable is created when Value() is called should return its value - 1]
[]interface {}{
    "value",
}
---

[Variable when a new variable is created when SetValue() is called should update the variable value - 1]
[]interface {}{
    "custom",
}
---

[Variable when a new variable is created when Name() is called should return the variable name - 1]
[]interface {}{
    "name",
}
---
